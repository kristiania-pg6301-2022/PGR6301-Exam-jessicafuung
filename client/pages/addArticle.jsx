import React, { useContext, useState } from "react";
import { ApiContext } from "../useContext";
import { useLoading } from "../useLoader";

export function AddNewArticle({ user }) {
  const { listArticles } = useContext(ApiContext);
  const { createArticle } = useContext(ApiContext);

  const [author, setAuthor] = useState("");
  const [title, setTitle] = useState();
  const [topic, setTopic] = useState("");
  const [articleText, setArticleText] = useState();

  const { loading, error, data } = useLoading(
    async () => await listArticles({ topic }),
    [topic]
  );

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  function handleSubmit(event) {
    event.preventDefault();
    createArticle({ author, title, topic, articleText });
  }

  if (user.microsoft !== undefined) {
    return (
      <>
        <form onSubmit={handleSubmit}>
          <h1>Skriv din artikkel, {user.microsoft.name}</h1>

          <ul>
            <p>Kategori:</p>
            <select value={topic} onChange={(e) => setTopic(e.target.value)}>
              <option value={"empty"}>Velg</option>
              <option value={"Politikk"}>Politikk</option>
              <option value={"Helse"}>Helse</option>
              <option value={"Natur"}>Natur</option>
              <option value={"Sport"}>Sport</option>
            </select>
          </ul>

          <ul>
            <p>Din overskrift: </p>
            <input
              required={"required"}
              value={title}
              onChange={(event) => setTitle(event.target.value)}
            />
          </ul>

          <ul>
            <p>Artikkel tekst: </p>
            <textarea
              required={"required"}
              value={articleText}
              onChange={(event) => setArticleText(event.target.value)}
            />
          </ul>

          <div>
            <button
              id={"finish-btn"}
              onClick={() => setAuthor(user.microsoft.name)}
            >
              Publiser
            </button>
          </div>
        </form>
      </>
    );
  }

  if (user.google !== undefined) {
    return <h1>You have no access to this action! Please login first</h1>;
  }

  if (!user || Object.keys(user).length === 0) {
    return <h1>You have no access to this action! Please login first </h1>;
  }
}
