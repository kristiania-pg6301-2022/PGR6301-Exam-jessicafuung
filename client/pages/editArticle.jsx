import React, { useContext, useEffect, useState } from "react";
import { ApiContext } from "../useContext";
import { useLoading } from "../useLoader";

let tmp = "";

export function EditArticle({ user }) {
  const { listArticles } = useContext(ApiContext);
  const { editArticle } = useContext(ApiContext);

  const [author, setAuthor] = useState("");
  const [oldTitle, setOldTitle] = useState("");
  const [newTitle, setNewTitle] = useState("");
  const [topic, setTopic] = useState("");
  const [articleText, setArticleText] = useState();

  const { loading, error, data } = useLoading(
    async () => await listArticles({ author }),
    [author]
  );

  const { name } = user.microsoft;

  useEffect(() => {
    if (user.microsoft !== undefined) {
      setAuthor(name);
    }
  }, [1]);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Du har ikke noen artikler Ã¥ redigere</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  function handleSubmit(event) {
    event.preventDefault();
    editArticle({ author, newTitle, oldTitle, topic, articleText });
  }

  if (author === name) {
    return (
      <>
        <form onSubmit={handleSubmit}>
          <ul>
            <p>Velg artikkel: </p>
            <select
              value={oldTitle}
              onChange={(e) => setOldTitle(e.target.value)}
            >
              <option value={"empty"}>Velg</option>
              {data.map((a) => {
                return (
                  <option key={a._id} value={a.value}>
                    {a.title}
                  </option>
                );
              })}
            </select>
          </ul>

          <ul>
            <p>Endre kategori:</p>
            <select value={topic} onChange={(e) => setTopic(e.target.value)}>
              <option value={"empty"}>Velg</option>
              <option value={"Politikk"}>Politikk</option>
              <option value={"Helse"}>Helse</option>
              <option value={"Natur"}>Natur</option>
              <option value={"Sport"}>Sport</option>
            </select>
          </ul>

          <ul>
            <p>Rediger overskrift: </p>
            <input
              required={"required"}
              value={newTitle}
              onChange={(event) => setNewTitle(event.target.value)}
            />
          </ul>

          <ul>
            <p>Rediger tekst: </p>
            <textarea
              required={"required"}
              value={articleText}
              onChange={(event) => setArticleText(event.target.value)}
            />
          </ul>

          <div>
            <button
              id={"finish-btn"}
              onClick={() => setAuthor(user.microsoft.name)}
            >
              Publiser
            </button>
          </div>
        </form>
      </>
    );
  } else {
    return <div></div>;
  }
}
