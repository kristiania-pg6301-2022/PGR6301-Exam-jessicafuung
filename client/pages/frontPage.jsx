import React, { useContext, useState } from "react";
import { ApiContext } from "../useContext";
import { useLoading } from "../useLoader";

export function ArticleCard({
  article: { title, topic, author, articleText },
}) {
  return (
    <>
      <ul>
        <h1>{title}</h1>{" "}
        <p style={{ color: "gray" }}>
          <u>
            Forfatter: {author} &nbsp; Kategori: {topic}
          </u>
        </p>
        <p> {articleText}</p>
      </ul>
    </>
  );
}

export function FrontPage({ user }) {
  const { listArticles } = useContext(ApiContext);
  const [topic, setTopic] = useState("");
  const [topicQuery, setTopicQuery] = useState("");
  const { loading, error, data } = useLoading(
    async () => await listArticles({ topic }),
    [topic]
  );

  function handleSubmitQuery(e) {
    e.preventDefault();
    setTopic(topicQuery);
  }

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  if (!user || Object.keys(user).length === 0) {
    return (
      <div>
        <h1>Velkommen til Fyrstikkposten!</h1>
        <h2>
          For å få tilgang til aktuelle nyhetsaker må du logge inn eller
          registrer deg for å få tilgang{" "}
        </h2>
      </div>
    );
  }

  if (user.google !== undefined || user.microsoft !== undefined) {
    return (
      <div>
        <div>
          <form onSubmit={handleSubmitQuery}>
            {data.map((article) => (
              <ul>
                <button
                  id="topic-query"
                  value={topicQuery}
                  onClick={(e) => setTopicQuery(article.topic)}
                >
                  {article.topic}
                </button>
              </ul>
            ))}
          </form>
        </div>
        {data.map((article) => (
          <ArticleCard key={article._id} article={article} />
        ))}
      </div>
    );
  }
}
