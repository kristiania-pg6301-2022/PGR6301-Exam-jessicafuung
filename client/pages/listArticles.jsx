import React, { useContext, useState } from "react";
import { useLoading } from "../useLoader";
import { ApiContext } from "../useContext";

/* display all articles */
function ArticlesCard({
  article: { author, title, topic, releaseDate, articleText },
}) {
  return (
    <>
      <div>
        <h1>{title}</h1>
        <h3>Written by: {author}</h3>
        <p>
          <b>Topic:</b> {topic}, <b>Date:</b> {releaseDate}
        </p>
        <div>{articleText}</div>
      </div>
    </>
  );
}

/* fetch from server and map it out */
export function ListArticles() {
  const { listArticles } = useContext(ApiContext);
  const [topic, setTopic] = useState("");
  const [topicQuery, setTopicQuery] = useState("");
  const { loading, error, data } = useLoading(
    async () => await listArticles({ topic }),
    [topic]
  );

  function handleSubmitQuery(e) {
    e.preventDefault();
    setTopic(topicQuery);
  }

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div>
      <h1>All News Articles in the database</h1>

      <div>
        <form onSubmit={handleSubmitQuery}>
          <label>
            Topic:
            <input
              id="topic-query"
              value={topicQuery}
              onChange={(e) => setTopicQuery(e.target.value)}
            />
            <button>Filter</button>
          </label>
        </form>
      </div>

      {data.map((article) => (
        <ArticlesCard key={article._id} article={article} />
      ))}
    </div>
  );
}
