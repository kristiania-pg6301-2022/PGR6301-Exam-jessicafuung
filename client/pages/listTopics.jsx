import React, { useContext, useEffect, useState } from "react";
import { ApiContext } from "../useContext";
import { useLoading } from "../useLoader";
import { Link, useNavigate } from "react-router-dom";

/* display topics in sidebar */
export function TopicsCard({ article: { _id, topic } }) {
  /* triggered when a clicked on a topic */
  function handleSubmit(event) {
    console.log("Handle submit: " + _id);
    event.preventDefault();
    setShowArticle(true);
    /*ShowSelectedArticle(_id);*/
  }

  return (
    <form onClick={handleSubmit}>
      <div>
        <ul>
          <Link to={"/news/selected"}>{topic}</Link>
        </ul>
      </div>
    </form>
  );
}

/* fetch from server  */
export function ListTopics() {
  const { listArticles } = useContext(ApiContext);
  const [topic, setTopic] = useState("");
  const { loading, error, data } = useLoading(
    async () => await listArticles({ topic }),
    [topic]
  );

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div>
      <h1 style={{ marginLeft: 40 }}>Topics: </h1>
      {data.map((article) => (
        <TopicsCard key={article._id} article={article} />
      ))}
    </div>
  );
}
