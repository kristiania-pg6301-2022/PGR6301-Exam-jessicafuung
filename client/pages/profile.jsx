import { Link } from "react-router-dom";
import React from "react";

export function Profile({ user }) {
  if (!user || Object.keys(user).length === 0) {
    return <h1>Please login first</h1>;
  }

  if (user.google !== undefined) {
    const { name, email, picture } = user.google;

    return (
      <div id={"profile-div"}>
        <h1>Profile for {name}</h1>
        <div>
          <img src={picture} />
          <p>Name: {name}</p>
          <p>Email: {email}</p>
        </div>
      </div>
    );
  }
  if (user.microsoft !== undefined) {
    const { name, email, picture } = user.microsoft;
    return (
      <div className={"CenterDiv"}>
        <h1>Profile</h1>
        <div>
          <img src={picture} />
          <p>Name: {name}</p>
          <p>Email: {email}</p>
        </div>
      </div>
    );
  }
}
