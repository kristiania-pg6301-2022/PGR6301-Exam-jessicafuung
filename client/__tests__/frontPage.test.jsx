import { FrontPage } from "../pages/frontPage";

import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import { Profile } from "../pages/profile";

const article = {
  title: "tittel test",
  topic: "topic test",
  author: "author test",
  articleText: "articleText test",
};

describe("Front page", () => {
  it("snapshot1", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FrontPage />
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });

  it("snapshot2", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
