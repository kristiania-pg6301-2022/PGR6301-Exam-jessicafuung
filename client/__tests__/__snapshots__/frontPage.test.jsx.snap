// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Front page snapshot1 1`] = `"<div>Loading...</div>"`;

exports[`Front page snapshot2 1`] = `"<h1>Vennligst logg inn f√∏rst</h1>"`;
